core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java

```diff
@@ -83,7 +84,59 @@ public void setByteArrayPage(byte[][] byteArray) {
   /**
    * Create a new column page based on the LV (Length Value) encoded bytes
    */
-  static ColumnPage newDecimalColumnPage(byte[] lvEncodedBytes) throws MemoryException {
+  static ColumnPage newDecimalColumnPage(byte[] lvEncodedBytes, int scale, int precision)
+      throws MemoryException {
+    DecimalConverterFactory.DecimalConverter decimalConverter =
+        DecimalConverterFactory.INSTANCE.getDecimalConverter(precision, scale);
+    int size = decimalConverter.getSize();
+    if (size < 0) {
+      return getLegacyColumnPage(lvEncodedBytes, scale, precision, DataType.DECIMAL);
+    } else {
+      // Here the size is always fixed.
+      return getDecimalColumnPage(lvEncodedBytes, scale, precision, size);
+    }
+  }
+
+  /**
+   * Create a new column page based on the LV (Length Value) encoded bytes
+   */
+  static ColumnPage newVarLengthColumnPage(byte[] lvEncodedBytes, int scale, int precision)
+      throws MemoryException {
+    return getLegacyColumnPage(lvEncodedBytes, scale, precision, DataType.BYTE_ARRAY);
```

Comment:
I think it is better to rename `LegacyColumnPage` to 'LVBytesColumnPage`. It is hard to know what is legacy